src/Service.Tests/GraphQLBuilder/Sql/StoredProcedureBuilderTests.cs

```diff
@@ -0,0 +1,234 @@
+                { spMutationEntityName, spMutationObjectTypeDefinition }
+            };
+
+            DocumentNode root = CreateGraphQLDocument(objectTypeDefinitions);
+
+            // Create permissions and entities collections used within the mutation and query builders.
+            _entityPermissions = GraphQLTestHelpers.CreateStubEntityPermissionsMap(
+                entityNames: new[] { spQueryEntityName, spMutationEntityName },
+                operations: new[] { Config.Operation.Execute },
+                roles: SchemaConverterTests.GetRolesAllowedForEntity()
+                );
+            Dictionary<string, Entity> entities = new()
+            {
+                { spMutationEntityName, spMutationEntity },
+                { spQueryEntityName, spQueryEntity }
+            };
+
+            try
+            {
+                // Build GraphQL schema document for the mutation which hydrates parameter metadata and
+                // attempts to convert the parameter value provided in configuration
+                // to the value type denoted in the database schema (metadata supplied via DatabaseObject).
+                DocumentNode mutationRoot = MutationBuilder.Build(
+                    root,
+                    DatabaseType.mssql,
+                    entities: entities,
+                    entityPermissionsMap: _entityPermissions,
+                    dbObjects: new Dictionary<string, DatabaseObject> { { spMutationEntityName, spDbObj } }
+                );
+
+                // Get mutation object type definition and validate its contents.
+                ObjectTypeDefinitionNode mutation = MutationBuilderTests.GetMutationNode(mutationRoot);
+                ValidateStoredProcedureSchema(typeNode: mutation, expectedGraphQLType, spMutationTypeName, parameterName);
+
+                // Build GraphQL schema document for the query which hydrates parameter metadata and
+                // attempts to convert the parameter value provided in configuration
+                // to the value type denoted in the database schema (metadata supplied via DatabaseObject).
+                DocumentNode queryRoot = QueryBuilder.Build(
+                    root,
+                    DatabaseType.mssql,
+                    entities: entities,
+                    inputTypes: null,
+                    entityPermissionsMap: _entityPermissions,
+                    dbObjects: new Dictionary<string, DatabaseObject> { { spQueryEntityName, spDbObj } }
+                );
+
+                // Get query object type definition and validate its contents.
+                ObjectTypeDefinitionNode query = QueryBuilderTests.GetQueryNode(queryRoot);
+                ValidateStoredProcedureSchema(typeNode: query, expectedGraphQLType, spQueryTypeName, parameterName);
+
+                Assert.IsFalse(expectsError, message: $"Failure: Expected a value type conversion error during stored procedure field creation.");
```

Comment:
We dont expectError here right? So, shouldnt the failure message here be Did NOT expect a value type conversion error?

Reply:
I'm trying to communicate that the conversion succeeded when it should not have. When expectsError is true, we should not even reach that assert. I believe the failure message should be accurate as is.

Reply:
I see what you mean now.